src/mono/wasm/Wasm.Build.Tests/PInvokeTableGeneratorTests.cs

```diff
@@ -375,6 +375,39 @@ public static int Main()
             Assert.Matches("warning\\sWASM0001.*Skipping.*Test::SomeFunction1.*because.*function\\spointer", output);
         }
 
+        [Theory]
+        [BuildAndRun(host: RunHost.None)]
+        public void UnmanagedCallback_InFileType_Compiles(BuildArgs buildArgs, string id)
+        {
+            string code =
+                """
+                using System;
+                using System.Runtime.InteropServices;
+                public class Test
+                {
+                    public static int Main()
+                    {
+                        Console.WriteLine("Main running");
+                        return 42;
+                    }
+                }
+
+                file class Foo
+                {
+                    [UnmanagedCallersOnly]
+                    public unsafe static extern void SomeFunction1(int i);
+                }
+                """;
+
+            (_, string output) = BuildForVariadicFunctionTests(
```

Comment:
We should run this also to ensure that any references to such symbols are correct, and usable.